File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.go
func numberOfSubstrings(str string) int {
	count := [3]int{}
	left := 0
	result := 0

	for right := 0; right < len(str); right++ {

		count[str[right]-'a']++

		for count[0] > 0 && count[1] > 0 && count[2] > 0 {
			result += len(str) - right

			count[str[left]-'a']--
			left++
		}
	}

	return result
}

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        Map<Character, Integer> map = new HashMap();
        int result = 0;
        map.put('a', -1);
        map.put('b', -1);
        map.put('c', -1);

        for(int i = 0; i < s.length(); i++)
        {
            char current = s.charAt(i);
            map.put(current, i);
            if(map.values().stream().allMatch(v -> v > -1)) //if the window contains all three
            {
                int minValue = map.values().stream().min(Integer::compareTo).orElse(0);//get the start point of the minimum window that contains all three
                result += minValue + 1;
            }
        }
        return result;
    }
}